### ISSUE TITLE
Clients only texting on off-topic and town-square

### ISSUE THREAD

**Issue by aspedrosa** [mattermost/mattermost-load-test-ng#247]
Currently I'm using the `go run ./cmd/ltagent --config config/config.json --controller-config config/simplecontroller.json -d 30` command to launch the tests.
Users, teams and channels are being created just fine but the clients only text on the off-topic and town-square channels of each team.
I'm using the follwing config
```
{
  "ConnectionConfiguration": {
    "ServerURL": "http://10.1.0.12:6300",
    "WebSocketURL": "ws://10.1.0.12:6300",
    "AdminEmail": "admin",
    "AdminPassword": "Adminadmin1."
  },
  "UserControllerConfiguration": {
    "Type": "simple",
    "Rate": 1.0
  },
  "InstanceConfiguration": {
    "NumTeams": 5,
    "NumChannels": 50,
    "NumTeamAdmins": 2,
    "TeamAdminInterval": 10
  },
  "UsersConfiguration": {
    "InitialActiveUsers": 50,
    "MaxActiveUsers": 1000,
    "AvgSessionsPerUser": 1
  },
  "LogSettings": {
    "EnableConsole": true,
    "ConsoleLevel": "INFO",
    "ConsoleJson": false,
    "EnableFile": true,
    "FileLevel": "INFO",
    "FileJson": true,
    "FileLocation": "ltagent.log"
  }
}
```

I'm using the default controller config.

I found some log lines like this
```
2020-04-23T18:28:18.070+0100	info	loadtest/loadtest.go:45	control.Login loadtest/control/actions.go:70 : No channels were found., 	{"controller_id": 46, "user_id": "qteiwujcatbrzd3oc6pzbufkrw"}
2020-04-23T18:28:18.070+0100	info	loadtest/loadtest.go:45	control/simplecontroller.(*SimpleController).reload loadtest/control/simplecontroller/actions.go:131 user is not connected	{"controller_id": 46, "user_id": "qteiwujcatbrzd3oc6pzbufkrw"}
2020-04-23T18:28:18.071+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 37, "user_id": "wddsxas4c3fiunkq61twuewaso"}
2020-04-23T18:28:18.093+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 35, "user_id": "a7qsdifuebykzyuizr65pik5zo"}
2020-04-23T18:28:18.098+0100	info	loadtest/loadtest.go:45	control.Login loadtest/control/actions.go:70 : No channels were found., 	{"controller_id": 47, "user_id": "47yourdcmp8qmkwm9pgptptyuh"}
2020-04-23T18:28:18.098+0100	info	loadtest/loadtest.go:45	control/simplecontroller.(*SimpleController).reload loadtest/control/simplecontroller/actions.go:131 user is not connected	{"controller_id": 47, "user_id": "47yourdcmp8qmkwm9pgptptyuh"}
2020-04-23T18:28:18.101+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 9, "user_id": "k915ujunajgp7b44csb3ds84cw"}
2020-04-23T18:28:18.103+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 44, "user_id": "4jymw3jdgj8r5fib8nxt87jb7o"}
2020-04-23T18:28:18.103+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 18, "user_id": "mp1yp3749pdfbxemc7m8nwfrxe"}
2020-04-23T18:28:18.104+0100	info	loadtest/loadtest.go:45	control.Login loadtest/control/actions.go:70 : No channels were found., 	{"controller_id": 12, "user_id": "jfgp5xig1jycfekbesaw8xz7ey"}
2020-04-23T18:28:18.104+0100	info	loadtest/loadtest.go:45	control/simplecontroller.(*SimpleController).reload loadtest/control/simplecontroller/actions.go:131 user is not connected	{"controller_id": 12, "user_id": "jfgp5xig1jycfekbesaw8xz7ey"}
2020-04-23T18:28:18.107+0100	info	loadtest/loadtest.go:52	skipping sending direct message	{"controller_id": 48, "user_id": "c8wz5m9y4tdc8kwooyu6zzccsh"}
2020-04-23T18:28:18.110+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 45, "user_id": "59dyf69j6bdffeyj5qjgn813bw"}
2020-04-23T18:28:18.114+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 31, "user_id": "87isahdwrfrdzqm9ejgtbusmqh"}
2020-04-23T18:28:18.114+0100	info	loadtest/loadtest.go:52	skipping sending direct message	{"controller_id": 34, "user_id": "aiq1wcmrsprz8k7h8pisfjqt5r"}
2020-04-23T18:28:18.123+0100	info	loadtest/loadtest.go:52	logged in	{"controller_id": 23, "user_id": "shuucnoi3bgstfc1azrseaa7ua"}
2020-04-23T18:28:18.133+0100	info	loadtest/loadtest.go:45	control.Login loadtest/control/actions.go:70 : No channels were found., 	{"controller_id": 3, "user_id": "mwsghe5habdddj9m75r6sway9a"}
2020-04-23T18:28:18.133+0100	info	loadtest/loadtest.go:45	control/simplecontroller.(*SimpleController).reload loadtest/control/simplecontroller/actions.go:131 user is not connected	{"controller_id": 3, "user_id": "mwsghe5habdddj9m75r6sway9a"}
```

Can somebody help? I don't know if this is the normal behavior or if there is any bug.

**Comment by grundleborg** [comment]
Hi @aspedrosa the load tests in this repository are still under active development and not yet ready for general usage. Can I ask what you are trying to use them for, and then I can probably direct you appropriately in terms of whether that is possible yet with this tool or whether you'd be better off using the existing load test tool at https://github.com/mattermost/mattermost-load-test

**Comment by aspedrosa** [comment]
The problem with mattermost-load-test is that it needs ssh access to the mattermost intances.
Currently I have a stack deployed on a docker swarm cluster where I can only access via docker.

**Comment by streamer45** [comment]
@aspedrosa 

As @grundleborg correctly pointed out , the project is still under heavy development and things might not work as expected. 
What I can say is that the `SimpleController` should not be used to do a proper simulation, in fact the reason users are posting only in the default channels is because we are not storing any other channel when joining a team.
The `SimulController` on the other hand does this but currently lacks an action to join a channel.
We are currently working on completing this controller so it will be improved quite soon.

**Comment by agnivade** [comment]
Hi @aspedrosa - how is it going with this ? Our recent development has been focused only on the `SimulController`. `SimpleController` is a very bare-bones implementation as of now. You might see better behavior using the SimulController.

All that said, this is still alpha software and very much in development. So there are still rough edges here and there.

**Comment by streamer45** [comment]
Closing this one for now. We've recently released [version 1.0](https://github.com/mattermost/mattermost-load-test-ng/releases) which should have fixed the reported issues.
